fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/DedupCursor.java

```diff
@@ -59,24 +59,25 @@
  */
 @API(API.Status.EXPERIMENTAL)
 public class DedupCursor<T> implements RecordCursor<T> {
-    // Inner cursor
+    /** Inner cursor. */
     @Nonnull
     private final RecordCursor<T> inner;
-    // The result returned from the cursor
+    /** The result returned from the cursor. */
     @Nullable
     private RecordCursorResult<T> nextResult;
-    // The last value found (to be compared against the next value generated)
+    /** The last value found (to be compared against the next value generated). */
     @Nullable
     private T lastValue;
-    // The method that can pack a value into a byte[]
+    /** The method that can pack a value into a byte[]. */
     @Nonnull
     private final Function<T, byte[]> packValue;
 
     /**
      * Constructor.
      * @param innerCursorFactory factory method to create an inner cursor given a continuation
      * @param unpackValue a method that can unpack a value from byte array (used to deserialize from a continuation)
-     * @param packValue a method that can pack a value into a byte array (used to serialize to a continuation)
+     * @param packValue a method that can pack a value into a byte array (used to serialize to a continuation). Note that
+     * this will only be called if the value is NOT null. Null value will be converted internally to null byte[]
```

Comment:
```suggestion
     * this will only be called if the value is NOT null. Null value will be converted internally to null byte[].
     * {@code pack} itself should never return {@code null}.
```